recyclerview/src/main/java/com/crazylegend/recyclerview/RecyclerViewExtensions.kt

```diff
@@ -628,4 +630,16 @@ fun RecyclerView.ViewHolder.shimmerAnimation(FADE_DURATION: Long = 1000L): Objec
         // Reset the alpha on animation end.
         doOnEnd { itemView.alpha = 1f }
     }
+}
+
+// Inspiration: https://github.com/sansets/android-clean-architecture
+fun RecyclerView.setAppBarElevationListener(appBar: AppBarLayout?, elevation: Float = 8f) {
+    addOnScrollListener(object : RecyclerView.OnScrollListener() {
+        override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
+            appBar?.let {
```

Comment:
Instead of adding a scroll listener if the app bar is null, above the addOnScrollListener you can simply
Add appBar?:return

That way we won't have a scroll listener, listening for nothing

Reply:
Yes, thank you

what do you mean like this?

fun RecyclerView.setAppBarElevationListener(appBar: AppBarLayout, elevation: Float = 8f) {
    addOnScrollListener(object : RecyclerView.OnScrollListener() {
        override fun onScrolled(recyclerView: RecyclerView, dx: Int, dy: Int) {
        if (canScrollVertically(-1)) ViewCompat.setElevation(it, elevation)
                else ViewCompat.setElevation(it, 0f)
            }
    })
}

Reply:
That's even better

Reply:
I was updating my PR